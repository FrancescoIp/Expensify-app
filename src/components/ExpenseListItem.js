import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, id}) => (
  <div>
    <li>
      <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
      <p>Amount: {amount/100}</p>
      <p>Created At: {createdAt}</p>
    </li>
  </div>
)


export default ExpenseListItem