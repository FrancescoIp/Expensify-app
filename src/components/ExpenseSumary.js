import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpense from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total';

export const ExpenseSumary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
  const total = numeral(expensesTotal / 100).format('$0,0.00')
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord}. Total: <span>{total}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  )
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpense(state.expenses, state.filters)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpenseTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpenseSumary);