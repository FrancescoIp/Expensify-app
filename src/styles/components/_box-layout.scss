.box-layot {
  background: url('/public/images/bg.jpg');
}