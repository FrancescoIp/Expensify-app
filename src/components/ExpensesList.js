import React from 'react';
import ExpenseListItem from './ExpenseListItem';
import { connect } from 'react-redux';
import selectExpense from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses to show</p>
      ) : (
          props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />
          })
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpense(state.expenses, state.filters)

  }
}

export default connect(mapStateToProps)(ExpenseList);

//Cosa succede veramente
// const ConnectedExpenseList = connect((state) => {
//   return {
//     expenses: state.expenses
//   };
// })(ExpensesList);

// export default ConnectedExpenseList