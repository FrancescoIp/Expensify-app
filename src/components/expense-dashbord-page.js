import React from 'react';
import ExpenseList from './ExpensesList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSumary from './ExpenseSumary';

const ExpensDashbordPage = () => (
  <div>
    <ExpenseSumary/>
    <ExpenseListFilters/>
    <ExpenseList/>
  </div>
);

export default  ExpensDashbordPage