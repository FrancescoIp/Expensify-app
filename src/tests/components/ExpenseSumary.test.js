import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSumary } from '../../components/ExpenseSumary';

test('Should render correctly expenseSumary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSumary expenseCount={1} expensesTotal={233}/>);
  expect(wrapper).toMatchSnapshot()
});

test('Should render correctly expenseSumary with 2 expenses', () => {
  const wrapper = shallow(<ExpenseSumary expenseCount={44} expensesTotal={232356623}/>)
  expect(wrapper).toMatchSnapshot()
});