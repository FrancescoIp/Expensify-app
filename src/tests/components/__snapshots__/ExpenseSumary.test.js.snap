// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly expenseSumary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
      . Total: 
      <span>
        $2.33
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render correctly expenseSumary with 2 expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        44
      </span>
       
      expenses
      . Total: 
      <span>
        $2,323,566.23
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
